airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -1412,6 +1431,31 @@ private JobRead waitUntilTheNextJobIsStarted(final UUID connectionId) throws Exc
     return mostRecentSyncJob;
   }
 
+  /**
+   * Waits for the given connection to finish, waiting at 30s intervals, until maxRetries is reached.
+   *
+   * @param connectionId the connection to wait for
+   * @param maxRetries the number of times to retry
+   * @throws InterruptedException
+   */
+  private void waitForSuccessfulJobWithRetries(final UUID connectionId, int maxRetries) throws InterruptedException {
```

Comment:
Why do we need a new function? `waitForSuccessfulJob` was not sufficient?

Reply:
In the past we saw issues with the jobs api so we added these retries; but in the longterm we'd like this to be unnecessary indeed.